src/Payments.sol

```diff
@@ -223,6 +231,56 @@ contract Payments is
         return railId;
     }
 
+    function terminateRail(
+        uint256 railId
+    )
+        external
+        validateRailActive(railId)
+        onlyRailClient(railId)
+        noRailModificationInProgress(railId)
+    {
+        Rail storage rail = rails[railId];
+
+        Account storage payer = accounts[rail.token][rail.from];
+        // Settle account lockup and ensure it's fully settled up to the current block
+        (bool fullySettled, uint256 settledEpoch) = settleAccountLockup(payer);
+        require(
+            fullySettled && settledEpoch == block.number,
+            "account lockup must be fully settled to terminate rail"
+        );
+
+        // If there is no arbiter, settle the rail immediately and ensure it is fully settled upto and including
+        // the current epoch
+        if (rail.arbiter == address(0)) {
+            (, uint256 settledUpto, ) = settleRail(railId, block.number);
+            require(
+                settledUpto == block.number,
+                "rail must be fully settled to terminate"
+            );
+        } else {
+            // For arbitrated rails, push the current rate to the queue
+            // This ensures proper settlement when the operator later settles this rail as we're setting the current rail rate to zero in this function
+            rail.rateChangeQueue.enqueue(rail.paymentRate, block.number);
+        }
+
+        // Update payer's lockup rate and current lockup
+        require(
+            payer.lockupRate >= rail.paymentRate,
+            "payer lockup rate must be at least rail payment rate"
+        );
+        payer.lockupRate = payer.lockupRate - rail.paymentRate;
+
+        // Set rail payment rate to 0 for future periods and mark the rail as inactive
+        rail.paymentRate = 0;
```

Comment:
We don't want to lock _new_ funds, but we still need to allow paying out the already locked funds up till the end of the lockup period.